activerecord/lib/active_record/railtie.rb

```diff
@@ -372,6 +372,14 @@ class Railtie < Rails::Railtie # :nodoc:
       end
     end
 
+    initializer "active_record.set_destroy_association_async_job" do
+      ActiveSupport.on_load(:active_job) do
+        ActiveSupport.on_load(:active_record) do
+          self.destroy_association_async_job ||= ActiveRecord::DestroyAssociationAsyncJob
+        end
+      end
```

Comment:
In the case where nothing else references / loads `ActiveJob::Base`, this will prevent `destroy_association_async_job` from being set at all.  For example:

```ruby
class Post < ApplicationRecord
  has_many :comments, dependent: :destroy_async
end

class Comment < ApplicationRecord
  belongs_to :post
end
```

```console
$ bin/rails runner "puts Post.first"
.../activerecord/lib/active_record/associations/builder/association.rb:132:in `check_dependent_options': ActiveJob is required to use destroy_async on associations (ActiveRecord::ActiveJobRequiredError)
```

I assume the tests do not catch it because `ActiveJob::Base` has already been loaded.  (Though the isolated suite might, but it is only run on a commit to `main` branch.)

We could instead check that `ActiveJob` (not `ActiveJob::Base`) is defined:

```suggestion
      ActiveSupport.on_load(:active_record) do
        if defined?(ActiveJob)
          self.destroy_association_async_job ||= ActiveRecord::DestroyAssociationAsyncJob
        end
      end
```

But this kind of approach seems to have been rejected in the original implementation of this feature: https://github.com/rails/rails/pull/36912#discussion_r400514794.  That thread also indicates that putting the load hook in `activejob/lib/active_job/railtie.rb` was related and intentional.  @rafaelfranca, do you still have objections to this approach?